src/EnergyPlus/SizingAnalysisObjects.hh

```diff
@@ -0,0 +1,191 @@
+#ifndef SizingAnalysisObjects_hh_INCLUDED
+#define SizingAnalysisObjects_hh_INCLUDED
+
+// C++ Headers
+#include <string>
+#include <vector>
+#include <map>
+
+// EnergyPlus Headers
+
+#include <DataLoopNode.hh>
+#include <WeatherManager.hh>
+#include <OutputProcessor.hh>
+#include <DataSizing.hh>
+#include <DataHVACGlobals.hh>
+#include <DataPlant.hh>
+#include <OutputReportPredefined.hh>
+#include <General.hh>
+#include <ObjexxFCL/gio.hh>
+
+namespace EnergyPlus {
+
+
+class SystemTimestepObject {
+public: 
+	Real64 CurMinuteStart		= 0.0; //minutes at beginning of system timestep
+	Real64 CurMinuteEnd			= 0.0; //minutes at end of system timestep
+	Real64 TimeStepDuration		= 0.0; //in fractional hours, length of timestep
+	Real64 LogDataValue			= 0.0; // raw hard value, obtained from pointer
+	int stStepsIntoZoneStep		= 0;
+};
+
+
+class ZoneTimestepObject {
+public:
+
+	int kindOfSim			= 0;
+	int envrnNum			= 0;
+	int dayOfSim			= 0;    // since start of simulation
+	int hourOfDay			= 0; 
+	int ztStepsIntoPeriod	= 0;    //count of zone timesteps into period
+	Real64 stepStartMinute	= 0.0;  //minutes at beginning of zone timestep
+	Real64 stepEndMinute	= 0.0;  //minutes at end of zone timestep
+	Real64 timeStepDuration = 0.0;  //in fractional hours, length of timestep
+	Real64 logDataValue		= 0.0;
+	Real64 runningAvgDataValue = 0.0;
+	bool hasSystemSubSteps	= false;
+	int numSubSteps = 0;
+	std::vector< SystemTimestepObject > subSteps; //nested object array for system timesteps inside here.
+
+	ZoneTimestepObject (
+		int kindSim,
+		int environmentNum,
+		int daySim,
+		int hourDay,
+		int stepEndMin,
+		Real64 timeStepDurat,
+		int numOfTimeStepsPerHour
+	);
+
+	ZoneTimestepObject();
+};
+
+class SizingLog {
+public:
+	int NumOfEnvironmentsInLogSet;
+	int NumOfDesignDaysInLogSet;
+	int NumberOfSizingPeriodsInLogSet;
+	std::map < int, int > ztStepCountByEnvrnMap; // key is the seed original Environment number, or index in the Environment Structure
+	std::map < int, int > envrnStartZtStepIndexMap; // key is the seed original Environment number, produces index in zone step vector where that period begins
+	std::map < int, int > newEnvrnToSeedEnvrnMap; // key is the new HVAC sim envrionment number, produces Seed environment number 
+	int NumOfStepsInLogSet; // sum of all zone timestep steps in log
+	int timeStepsInAverage; // breadth back in time for running average, zone timesteps
+	Real64 *p_rVariable;    // pointer to variable being loggged
```

Comment:
I think that particular int is the second argument.  But yes I have seen that error on the CI mac's and suspect this is what is causing it, but I don't have the c++ to fix it.  

why a pointer? Attempting to create a general system for recording.  I could hard code the handful of variables I need to track for this project. But the aim was to make a reuseable logging framework that could be expanded for use with other sizing issues in the future.   It is the same approach used for output variables and seemed to me like a natural situation to use pointers.